torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -673,257 +669,72 @@ def inner_fn(args: List[Any]):
+            leaf_flat_args.append(a.detach().requires_grad_(a.requires_grad))
+        else:
+            ok = False
+            break
+
+    if ok:
+        return leaf_flat_args
+
+    if requires_subclass_dispatch(leaf_flat_args, fw_metadata):
+        raise RuntimeError(
+            """\
+    Encountered duplicate inputs that are mutated in the graph, but at least one input/output
+    to the graph is a tensor subclass. This is not supported today. You can try to
+    remove the aliasing yourself as a workaround, or otherwise file an issue on github."""
+        )
 
-        @wraps(compiled_fn)
-        def wrapped_compiled_fn(args: List[Any]):
-            deduped_args = self.remove_dupe_args(args)
-            args.clear()
-            return compiled_fn(deduped_args)
-
-        wrapped_compiled_fn._boxed_call = True  # type: ignore[attr-defined]
-
-        # This can be uncommented when we properly guard for duplicates,
-        # but right now we must not do it.
-        # if not config.debug_assert:
-        #     return wrapped_compiled_fn
-
-        @wraps(wrapped_compiled_fn)
-        def debugged_compiled_fn(args):
-            # Test that the computed remove/add arg functions are an inverse
-            new_args = self.add_dupe_args(self.remove_dupe_args(args))
-            seen: Dict[Any, None] = {}
-            for i, (x, y) in enumerate(zip(new_args, args)):
-                seen[y] = None
-                assert x is y, format_guard_bug_msg(
-                    aot_config,
-                    f"{describe_input(i, aot_config)} would be a duplicate of "
-                    f"{describe_input(self.add_dupe_map[i], aot_config)}",
-                )
-            # This is only an error if there is metadata mutation on both of
-            # the duped arguments; in this case, we need to know what order
-            # the metadata mutation applies in.  You'll get the correct result
-            # otherwise, because a graph that assumes distinct inputs works if
-            # you dupe the inputs (the gradient contributions from each input
-            # will get summed up appropriately.)
-            #
-            # TODO: work out how to setup this assert correctly
-            """
-            assert len(seen) == unique_args, format_guard_bug_msg(aot_config,
-                f"there would be {unique_args} distinct arguments"
-            )
-            """
-            return wrapped_compiled_fn(args)
+    # export path: ban duplicate inputs for now, add later if requested.
+    if aot_config.is_export:
+        raise RuntimeError(
+            f"""\
+    Encountered duplicated inputs that are mutated in the graph you are trying to export.
+    This functionality is currently not supported. If needed, please file a github issue.
 
-        debugged_compiled_fn._boxed_call = True  # type: ignore[attr-defined]
+    fw_metadata={str(fw_metadata)}
+        """
+        )
 
-        return debugged_compiled_fn
+    raise RuntimeError(
```

Comment:
This is the new error we throw if autograd encounters duplicate arguments that it doesn't know how to handle. Theoretically, this never runs when dynamo is involved.